src/query/storage/m3/storage.go

```diff
@@ -257,6 +257,8 @@ func (s *m3storage) Write(
 	id := query.Tags.ID()
 	// TODO: Consider caching id -> identID
```

Comment:
Caching these in an LRU would actually probably save a ton of allocs unrelated to this P.R

Reply:
I think there's some other refactoring of this going on in https://github.com/m3db/m3/pull/942, so don't want to step on any toes there and lead to messy merges; we'll definitely circle back on adding an LRU here